### PR TITLE
Check analysis dependencies for null

### LINKED ISSUE #1341: Argument null exception in PythonAnalyzerEntry.Invalidate
`UpdateAnalysisTcs` can set `_analysisDependencies` to null. In one `Invalidate` the null check is like
```cs
    UpdateAnalysisTcs(analysisVersion);
    dependencies = _analysisDependencies != null ? ...
```
but in another overload it is
```cs
    UpdateAnalysisTcs(analysisVersion);
    dependencies = _parserDependencies != null ? ...
```
which causes argument exception in `_parserDependencies.Union(_analysisDependencies)`

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
Fixes #1341

### CODE REVIEW

File: src/Analysis/Ast/Impl/Analyzer/PythonAnalyzerEntry.cs [hunk 1]
```diff
@@ -219,9 +218,14 @@ public Task<IDocumentAnalysis> GetAnalysisAsync(CancellationToken cancellationTo
...
 
                 UpdateAnalysisTcs(analysisVersion);
-                dependencies = _parserDependencies != null
-                    ? ImmutableArray<AnalysisModuleKey>.Create(_parserDependencies.Union(_analysisDependencies).ToArray())
-                    : ImmutableArray<AnalysisModuleKey>.Create(_analysisDependencies);
+                if (_analysisDependencies != null) {
+                    dependencies = _parserDependencies != null
+                        ? ImmutableArray<AnalysisModuleKey>.Create(_parserDependencies.Union(_analysisDependencies).ToArray())
+                        : ImmutableArray<AnalysisModuleKey>.Create(_analysisDependencies);
+                } else if(_parserDependencies != null) {
```
Comment (jakebailey) [inline src/Analysis/Ast/Impl/Analyzer/PythonAnalyzerEntry.cs]: Nit: formatting

Comment (MikhailArkhipov) [reply src/Analysis/Ast/Impl/Analyzer/PythonAnalyzerEntry.cs]: whats wrong with it?

Comment (jakebailey) [reply src/Analysis/Ast/Impl/Analyzer/PythonAnalyzerEntry.cs]: `if(` vs `if (`. Minor (but VS normally fixes it).

**Comment by AlexanderSher** [comment]
I think it would be better to remove `_analysisDependencies = null` from `UpdateAnalysisTcs`, cause it is already set to `null` in `TrySetAnalysis`.

**Comment by MikhailArkhipov** [comment]
Should we then assume non-null in 
```cs
public bool Invalidate(IPythonModule module, PythonAst ast, int bufferVersion, int analysisVersion, out ImmutableArray<AnalysisModuleKey> dependencies)
```
and just do
```
dependencies = ImmutableArray<AnalysisModuleKey>.Create(_parserDependencies.Union(_analysisDependencies).ToArray())
```
?

**Comment by AlexanderSher** [comment]
No, this is true only for library analysis. User code analysis erases all dependencies cause changed code may not contain both parsing-created dependencies and analysis-created dependencies.

**Pull request merged**